_posts/2018-03-12-roadmap.md

```diff
@@ -0,0 +1,234 @@
+
+### Web site improvements
+
+As part of Rust 2018, we will completely overhaul the Rust web site, making it
+useful for CTOs and engineers alike. It should be far easier to find information
+to help evaluate Rust for your use case, and to stay up to date with the latest
+tooling and ecosystem improvements.
+
+### Four target domains
+
+Part of our goal with Rust 2018 is to demonstrate Rust’s productivity in
+specific domains of use. We’ve selected four such domains to invest in and
+highlight this year:
+
+- **Network services**. Rust’s reliability and low footprint make it an
+  excellent match for network services and infrastructure, especially at high
+  scale.
+- **[Command-line apps]** (CLI). Rust’s portability, reliability, ergonomics, and ability to
+  produce static binaries come together to great effect for writing CLI apps.
+- **[WebAssembly]**. The “wasm” web standard allows shipping native-like binaries
+  to all major browsers, but GC support is still years away. Rust
+  is [extremely well positioned](https://mgattozzi.com/rust-wasm) to target this
+  domain, and provides a reasonable on-ramp for programmers coming from JS.
+- **[Embedded devices]**. Rust has the potential to make programming
+  resource-constrained devices much more productive—and fun! We want embedded
+  programming to reach first-class status this year.
+
+[Command-line apps]: https://internals.rust-lang.org/t/announcing-the-cli-working-group/6872
+[Embedded devices]: https://internals.rust-lang.org/t/announcing-the-embedded-devices-working-group/6839
+[WebAssembly]: https://internals.rust-lang.org/t/come-join-the-rust-and-webassembly-working-group/6845
+
+Each of these domains has a dedicated working group for the year. These WGs will
+work in a cross-cutting fashion, interfacing with language, tooling, library,
+and documentation work.
+
+### Compatibility across editions
+
+**TL;DR: Rust will continue its stability guarantee
+of [hassle-free updates to new versions][stability]**.
+
+[stability]: https://blog.rust-lang.org/2014/10/30/Stability.html
+
+Editions will have a meaning for the compiler. You will be able to write:
+
+```toml
+edition = "2018"
+```
+
+in your Cargo.toml to _opt in_ to the new edition for your crate. Doing so may
+introduce new keywords or otherwise require adjustments to code. However:
+
+-   You can use _old_ editions indefinitely on _new_ compilers; **editions are
+    opt-in**.
+-   Editions are set on a _per-crate_ basis and can be mixed and matched; **you
+    can be on a different edition from your dependencies**.
+-   Warning-free code in one edition must compile, and have the same behavior, on
+    the next.
+-   Edition-related warnings, e.g. that an identifier will become a keyword in the
+    next edition, must be easily fixable via an automated migration tool
+    (rustfix). **Only a small minority of crates should require _any_
+    manual work to opt in to a new edition**, and that manual work must be
```

Comment:
IMO it's worth mentioning in this list that many of the new features that are a part of Rust 2018 will be available on older editions too.